File: nodejs/cucumber/features/step_definitions/calculator_steps.js
const assert = require('assert');
const { Given, When, Then } = require('@cucumber/cucumber');
const Calculator = require('../../model/calculator');


Given("I have a calculator", () => {
  this.calculator = new Calculator();
});


When("I add {int} and {int}", (x, y) => {
  this.sum = this.calculator.add(x, y);
});


Then("The calculator returns {int}", (sum) => {
  /*this.epic("Cucumber");
  this.feature("Calculator");
  this.story("Addition");
  this.parentSuite("Cucumber");
  this.suite("Calculator");*/
  assert.equal(this.sum, sum);
});

File: python/cucumber/steps/calculator_steps.py
from behave import given, when, then
from model.calculator import Calculator


@given('I have a calculator')
def step_impl(context):
    context.calculator = Calculator()


@when('I add "{x:d}" and "{y:d}"')
def step_impl(context, x, y):
    assert isinstance(x, int)
    assert isinstance(y, int)
    context.calculator.add(x, y)


@then('the calculator returns "{result:d}"')
def step_impl(context, result):
    assert isinstance(result, int)
    assert context.calculator.result == result
